source/run-command/run-command.rst

```diff
@@ -187,8 +193,144 @@ Drivers MUST document the behavior of RunCommand if a ``maxTimeMS`` field  is al
 * See Client Side Operations Timeout's section on `runCommand <https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst#runcommand>`_
 * See Client Side Operations Timeout's section on `runCommand behavior <https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst#runcommand-behavior>`_
 
+
+--------------------
+``runCursorCommand``
+--------------------
+
+Drivers MAY expose a runCursorCommand API with the following syntax.
+
+.. code:: typescript
+
+    interface Database {
+      /**
+       * Takes an argument representing an arbitrary BSON document and executes it against the server.
+       */
+      runCursorCommand(command: BSONDocument, options: RunCursorCommandOptions): RunCommandCursor;
+    }
+
+    interface RunCursorCommandOptions extends RunCommandOptions {
+      /**
+       * For operations that create cursors, timeoutMS can either cap the lifetime of the cursor or be applied separately to the original operation and all subsequent calls.
+       * To support both of these use cases, these operations MUST support a timeoutMode option.
+       *
+       * @defaultValue 'cursorLifetime'
+       *
+       * @see https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst
+       */
+      timeoutMode?: 'iteration' | 'cursorLifetime';
+
+      /**
+       * Identifies the type of cursor this is.
+       *
+       * A tailable cursor can receive empty `nextBatch` arrays in `getMore` responses.
+       * However, subsequent `getMore` operations may return documents if new data has become available.
+       *
+       * A tailableAwait cursor is an enhancement where instead of dealing with empty responses the server will block until data becomes available.
+       *
+       * @defaultValue 'nonTailable'
+       */
+      cursorType: 'nonTailable' | 'tailable' | 'tailableAwait';
+    }
+
+    /**
+     * The following are the configurations a driver MUST provide to control how getMores are constructed.
+     * How the options are controlled should be idiomatic to the driver's language.
+     * See Executing ``getMore`` Commands.
+     */
+    interface RunCursorCommandGetMoreOptions {
+      batchSize?: PositiveIntegerNumber;
+      maxTimeMS?: PositiveIntegerNumber;
```

Comment:
Looking at the [CRUD spec](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst) for guidance here, I think we can just use Int32 and Int64 for `batchSize` and `maxTimeMS`, respectively. In the CRUD spec, `getMore` `maxTimeMS` is named `maxAwaitTimeMS` and uses an Int64. I don't think the server actually enforces Int32 for `batchSize` -- it's more likely that a batch size beyond `INT32_MAX` is just unreasonable due to wire protocol limits.

Doc block comments on these fields can communicate that `batchSize` is a positive integer and `maxTimeMS` is a non-zero integer, but I don't think we need to communicate that in the type here -- that's not done in other specs AFAIK.

Reply:
I'm starting to think it is worth surveying the use of bit widths in the specs where they are not actually related to a server requirement. For these fields any bit width is accepted as long as it does not have a fractional component. 

As a JS engineer we should be able to send non-fractional doubles, which is the native numeric type to our language, but the spec suggests otherwise in all these interfaces. 

Maybe we should start using number classes? for now I'll follow the docs lead here on phrasing
```ts
  /** Any natural number is permitted. */
  batchSize?: N;
  /** Zero and any natural number is permitted. */
  maxTimeMS?: N | 0;
```

Reply:
> I'm starting to think it is worth surveying the use of bit widths in the specs where they are not actually related to a server requirement. For these fields any bit width is accepted as long as it does not have a fractional component.

Feel free to create a DRIVERS ticket to track that.

> As a JS engineer we should be able to send non-fractional doubles, which is the native numeric type to our language, but the spec suggests otherwise in all these interfaces.

As far as the public API is concerned, I don't see any problem with accepting a non-fractional double if that's idiomatic for the language. So long as it's a modeled option, the driver can still ensure it's encoded to the correct BSON type if required by the server.

That said, using non-fractional doubles in a command document, which isn't inspected by the driver, is another issue entirely.